src/project/manifest/document.rs

```diff
@@ -285,6 +310,47 @@ impl ManifestSource {
         Ok(())
     }
 
+    /// Adds an environment to the manifest
+    pub fn add_environment<'f>(
+        &mut self,
+        name: &str,
+        features: impl IntoIterator<Item = &'f FeatureName>,
+        solve_group: Option<&str>,
+        no_default_features: bool,
+    ) -> Result<(), TomlError> {
+        let features = Array::from_iter(features.into_iter().map(|f| f.to_string()));
+
+        // Construct the TOML item
+        let item = if solve_group.is_some() || no_default_features {
+            let mut table = toml_edit::InlineTable::new();
+            table.insert("features", features.into());
```

Comment:
```suggestion
	if let Some(features) = features {
            table.insert("features", features.into());
        }
```
It's allowed to not define features. As you might want to have a different environment with the default set but solve grouped with another environment